### DISCUSSION TITLE
How to render virtualized gallery?

### DISCUSSION THREAD

**Discussion by lironezra** [TanStack/virtual#295] [Q&A]
### Discussed in https://github.com/TanStack/react-virtual/discussions/279

<div type='discussions-op-text'>

<sup>Originally posted by **lironezra** April 10, 2022</sup>
Hi friends :)

I'm trying to build a gallery with items like the image below I am using react-query to fetch the items with infinite-scrolling, 
I want to achieve that with virtualized items.
Can I implement it with Raws (I've tried this approach without success...) or I need to use the Grid view?
 
![image](https://user-images.githubusercontent.com/28096499/162609897-d6d4a487-48d0-44ec-b5a9-170314d3a051.png)

Thanks for the help!
</div>

**Comment by em2724** [comment]
Hi, did you make any progress on this? I'm currently making a similar gallery and using the grid view as in the docs example. But because the columns aren't actually virtualised I'm thinking of changing to only virtualising the rows and wrapping each row in a div. Simpler to layout out the items in a row using flex-box or css grid. Did you continue using the grid view?

**Comment by hvdson** [comment]
any solutions to this? i'm working on a similar view with infinite scroll as well but the columns need to be responsive

**Reply by piecyk** [reply]
So what exactly is the issue here? You want to virtualize rows and columns, did you check the grid example here https://tanstack.com/virtual/v3/docs/examples/react/dynamic

**Comment by AlexAdvent** [comment]
Has any one implemented this?

**Comment by wan54** [comment]
For anyone looking for the answer this is the example, specifically the masonry, you should check out  https://tanstack.com/virtual/latest/docs/framework/react/examples/variable.

**Comment by dango0812** [comment]
@wan54 @AlexAdvent 
I've had a similar experience myself, and based on that, I'm currently working on an example using grid and ratio grid layout.
https://github.com/dango0812/tanstack-virtual-grid

**Reply by piecyk** [reply]
Nice! Looking forward to seeing what you come up with.

**Reply by dango0812** [reply]
You can check out the design used TanStack Virtual at https://tanstack-virtual-grid.vercel.app/.

It supports both a 1:1 & 16:9 aspect ratio!
I'll also be sharing an example that shows how to implement it with React Query and infinite scrolling.

**Reply by piecyk** [reply]
Sweet!
